src/wfs/src/main/java/org/geoserver/wfs/kvp/FilterKvpParser.java

```diff
@@ -81,7 +82,9 @@ public Object parse(String value) throws Exception {
                 InputStream input = new ByteArrayInputStream(string.getBytes());
 
                 try {
-                    Filter filter = (Filter) parser.parse(input);
+                    InputSource inputSource = new InputSource(input);
+                    inputSource.setEncoding(Charset.defaultCharset().name());
```

Comment:
Hi Ben, checking the GeoServerInfo.getSettings() you will find a "getCharset()" method that returns "the default charset used for responses":
http://docs.geoserver.org/stable/en/user/webadmin/server/globalsettings.html#character-set

Makes me wonder if the best choice for the input should be the system one, or the one that's configured in GeoServer, for consistency sake between input and output.